core/src/test/java/org/elasticsearch/cluster/metadata/WildcardExpressionResolverTests.java

```diff
@@ -125,6 +126,50 @@ public void testAll() {
         assertThat(newHashSet(resolver.resolve(context, Arrays.asList("_all"))), equalTo(newHashSet("testXXX", "testXYY", "testYYY")));
     }
 
+    // WildcardExpressionResolver must be able to resolve the provided wildcard only against the
+    // defined indices when ignoreAliases option is set or against the defined indices and aliases
+    // otherwise
+    public void testConcreteIndicesWildcardAndAliases() {
+        MetaData.Builder mdBuilder = MetaData.builder()
+                .put(indexBuilder("foo_foo").state(State.OPEN).putAlias(AliasMetaData.builder("foo")))
+                .put(indexBuilder("bar_bar").state(State.OPEN).putAlias(AliasMetaData.builder("foo")));
+        ClusterState state = ClusterState.builder(new ClusterName("_name")).metaData(mdBuilder).build();
+
+        IndicesOptions indicesAndAliasesOptions = IndicesOptions.fromOptions(false, false, true, false, true, false, false);
+        IndexNameExpressionResolver.Context context = new IndexNameExpressionResolver.Context(state, indicesAndAliasesOptions);
+
+        IndicesOptions ignoreAliasesOptions = IndicesOptions.fromOptions(false, false, true, false, true, false, true);
+        IndexNameExpressionResolver.Context contextIndices = new IndexNameExpressionResolver.Context(state, ignoreAliasesOptions);
+
+        assertThat(
+                IndexNameExpressionResolver.WildcardExpressionResolver
+                        .matches(context, state.getMetaData(), "*").keySet(),
+                equalTo(newHashSet("bar_bar", "foo_foo", "foo")));
+        assertThat(
+                IndexNameExpressionResolver.WildcardExpressionResolver
+                        .matches(contextIndices, state.getMetaData(), "*").keySet(),
+                equalTo(newHashSet("bar_bar", "foo_foo")));
+
+        assertThat(
+                IndexNameExpressionResolver.WildcardExpressionResolver
+                        .matches(context, state.getMetaData(), "foo*").keySet(),
+                equalTo(newHashSet("foo", "foo_foo")));
+        assertThat(
+                IndexNameExpressionResolver.WildcardExpressionResolver
+                        .matches(contextIndices, state.getMetaData(), "foo*").keySet(),
+                equalTo(newHashSet("foo_foo")));
+
+        assertThat(
+                IndexNameExpressionResolver.WildcardExpressionResolver
+                        .matches(context, state.getMetaData(), "f*o").keySet(),
+                equalTo(newHashSet("foo", "foo_foo")));
+        assertThat(
+                IndexNameExpressionResolver.WildcardExpressionResolver
+                        .matches(contextIndices, state.getMetaData(), "f*o").keySet(),
+                equalTo(newHashSet("foo_foo")));
+
+    }
+
```

Comment:
would you mind adding a couple of integration tests to `IndexAliasesIT` also? Maybe test both IndicesAliasesRequest and GetAliasesRequest ?